tests/modeladmin/models.py

```diff
@@ -43,3 +43,45 @@ def decade_published_in(self):
 
 class ValidationTestInlineModel(models.Model):
     parent = models.ForeignKey(ValidationTestModel, models.CASCADE)
+
+
+class Professor(models.Model):
+    name = models.CharField(max_length=255)
+
+    def __str__(self):
+        return self.name
+
+
+class Lecture(models.Model):
+    name = models.CharField(max_length=255)
+    professor = models.ForeignKey(Professor, models.CASCADE)
+
+    def __str__(self):
+        return self.name
+
+
+class Student(models.Model):
+    name = models.CharField(max_length=255)
+    lectures = models.ManyToManyField(Lecture)
+    mentor = models.ForeignKey(Professor, models.CASCADE)
+
+    def __str__(self):
+        return self.name
+
+
+class Album(models.Model):
```

Comment:
Did you try to reuse existing models as we've mentioned before (adding more models slows down tests)? `ValidationTestInlineModel` has a foreign key, I'd think that could work.

Reply:
Oh, sorry. I will remove them all.